### PR TITLE
tensor.zeros to use async memset

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Addressed outstanding FIXME note to use async call to populate array with zeros.

Added optimization for `_full_usm_ndarray` to use `handler::memset` instead of `handler::fill` for 1-byte wide types and for other types when fill value is bitwise zero.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

**Comment by ndgrigorian** [comment]
@oleksandr-pavlyk 
Seems a few array API tests for full regressed with this change, we should see if it's a result of the changes or independent.

File: dpctl/tensor/libtensor/source/full_ctor.cpp [hunk 1]
```diff
@@ -80,10 +80,65 @@ sycl::event full_contig_impl(sycl::queue &exec_q,
...
+            cgh.depends_on(depends);
+
+            cgh.memset(reinterpret_cast<void *>(dst_p), memset_val,
+                       nelems * sizeof(dstTy));
+        });
+    }
+    else {
+        bool is_zero = false;
+        if constexpr (sizeof(dstTy) == 1) {
+            is_zero = (std::uint8_t{0} == sycl::bit_cast<std::uint8_t>(fill_v));
```
Comment (ndgrigorian) [inline dpctl/tensor/libtensor/source/full_ctor.cpp:97]: This is cool, I hadn't realized some C++20 features were available in the `sycl` namespace.

File: dpctl/tensor/libtensor/source/zeros_ctor.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+static zeros_contig_fn_ptr_t zeros_contig_dispatch_vector[td_ns::num_types];
+
+std::pair<sycl::event, sycl::event>
+usm_ndarray_zeros(const dpctl::tensor::usm_ndarray &dst,
+                  sycl::queue &exec_q,
+                  const std::vector<sycl::event> &depends)
+{
+    // start, end should be coercible into data type of dst
+
```
Comment (ndgrigorian) [inline dpctl/tensor/libtensor/source/zeros_ctor.cpp:106]:
```suggestion
```

**Review by ndgrigorian — APPROVED** [review]
Approved, the change looks good to me

File: dpctl/tensor/libtensor/source/full_ctor.cpp (hunk 1 above)
Comment (oleksandr-pavlyk) [reply dpctl/tensor/libtensor/source/full_ctor.cpp:97]: SYCL spec documents features adopted from future revisions of C++ standard: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:progmodel.futurecppversion

**Pull request merged**